### PR TITLE
fix(plan): improve plan module test coverage and fix hint bug

### PR DESCRIPTION
**Pull request by WangCheng0116**
## Summary

This PR improves the test coverage for the plan module and fixes a critical bug in hint generation:

- **Fix hint generation bug**: Corrected the condition ordering in `DefaultPlanToHint.__call__` to ensure proper hint generation when all subtasks are completed
- **Add comprehensive tests**: Added 171 lines of new tests covering previously untested code paths in `_plan_notebook.py`
- **Add toolkit tests**: Added unit tests for `json_schema` registration in toolkit

### 1. Bug Fix (src/agentscope/plan/_plan_notebook.py)
- Fixed condition ordering in `DefaultPlanToHint.__call__`
- Previously, when all subtasks were done, the wrong hint (`when_no_subtask_in_progress`) was shown instead of `at_the_end`
- Moved the "all done/abandoned" check before the "partial done" check

### 2. New Tests (tests/plan_test.py)
Added 3 comprehensive test methods:

- `test_hint_generator_all_states`: Tests all 6 state combinations in DefaultPlanToHint
  - No plan, all todo, in progress, partial done, all done, mixed done/abandoned
- `test_error_paths`: Tests error handling paths
  - Invalid types, invalid actions, out-of-bounds indices, invalid states
  - State transition constraints
- `test_edge_cases`: Tests edge cases
  - Plan replacement, auto-activation of next subtask
  - Historical plan operations, hook mechanisms

### 3. Toolkit Tests (tests/toolkit_basic_test.py)
- Added unit tests for `json_schema` registration

## Checklist

Please check the following items before code is ready to be reviewed.

- [x]  Code has been formatted with `pre-commit run --all-files` command
- [x]  All tests are passing
- [x]  Docstrings are in Google style
- [x]  Related documentation has been updated (e.g. links, examples, etc.)
- [x]  Code is ready for review

### CODE REVIEW

**Comment by WangCheng0116** [comment]
Hi team, any feedback on this PR would be appreciated. This is my first contribution here, and I’m happy to make any necessary changes.

**Comment by DavdGao** [comment]
> Hi team, any feedback on this PR would be appreciated. This is my first contribution here, and I’m happy to make any necessary changes.

Thanks for opening this PR and for your first contribution! We'll take a look and provide feedback shortly.

File: src/agentscope/plan/_plan_notebook.py [hunk 1]
```diff
@@ -356,21 +356,25 @@ async def revise_current_plan(
...
 
-        # validate subtask_idx
-        if action != "add" and subtask_idx >= len(self.current_plan.subtasks):
-            response.append(
-                f"Invalid subtask_idx '{subtask_idx}' for action '{action}'. "
-                f"Must be between 0 "
-                f"and {len(self.current_plan.subtasks) - 1}.",
-            )
+        # validate subtask_idx range only when its type is valid
+        if isinstance(subtask_idx, int):
```
Comment (DavdGao) [inline src/agentscope/plan/_plan_notebook.py]: Maybe we just raise the type error in line 336 rather than add this conditional statement?

Comment (WangCheng0116) [reply src/agentscope/plan/_plan_notebook.py]: fixed, could u plz check again?

**Review by DavdGao — APPROVED** [review]
